frontend/ui/uimanager.lua

```diff
@@ -218,6 +219,8 @@ function UIManager:close(widget, refreshtype, refreshregion)
     local dirty = false
     -- first send close event to widget
     widget:handleEvent(Event:new("CloseWidget"))
+    -- Ensure all the widgets can get onFlushSettings event.
+    widget:handleEvent(Event:new("FlushSettings"))
```

Comment:
Yes, this change won't fix anything in the current code base. I am more worried about bugs introduced by future developers assuming flushsettings event will always happen before close.